### PR TITLE
Fix the handling of custom names (AssemblyName, PackageId) in packages lock file, construction and validation.

### LINKED ISSUE #7889: RestoreLockedMode: Unexpected NU1004 when ProjectReference refers to a project with custom AssemblyName
_From @patrikbeno on March 17, 2019 12:23_

## Steps to reproduce

```xml
<!-- ClassLib1/ClassLib1.csproj -->
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
    <!-- following line breaks RestoreLockedMode=true -->
    <AssemblyName>CustomName</AssemblyName>
  </PropertyGroup>
</Project>

<!-- ClassLib2/ClassLib2.csproj -->
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <ProjectReference Include="..\ClassLib1\ClassLib1.csproj" />  
  </ItemGroup>
</Project>
```

```
$ cd ClassLib2
$ dotnet restore -f --use-lock-file
$ dotnet restore -f --locked-mode
```

## Expected  behavior

`dotnet restore -f --locked-mode` completes without error.

## Actual behavior

```
ClassLib2/ClassLib2.csproj : error NU1004: The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Please disable RestoreLockedMode MSBuild property or pass explicit --force-evaluate flag to run restore to update the lock file.
Restore failed in 56.35 ms for /solution/ClassLib2/ClassLib2.csproj.
```

## Environment data
`dotnet --info` output:

tested in latest `dotnet/core/sdk:2.2-alpine` docker image

```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.105
 Commit:    7cecb35b92

Runtime Environment:
 OS Name:     alpine
 OS Version:  3.8
 OS Platform: Linux
 RID:         linux-musl-x64
 Base Path:   /usr/share/dotnet/sdk/2.2.105/

Host (useful for support):
  Version: 2.2.3
  Commit:  6b8ad509b6
```

_Copied from original issue: dotnet/cli#10988_

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/nuget/home/issues/7889
Regression: No, Never worked
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 
The comparison now uses the short project name which is what's used when writing the lock file. 

## Testing/Validation

Tests Added: Yes  
Reason for not adding tests:  
Validation:

### CODE REVIEW

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs [hunk 1]
```diff
@@ -7405,5 +7405,153 @@ public async Task RestoreNetCore_IncompatiblePackageTypesFailRestore()
...
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    packageX);
+
+                // A -> B
+                projectA.Properties.Add("RestorePackagesWithLockFile", "true");
+                projectB.Properties.Add("AssemblyName", "CustomName");
+
+                //projectA.Properties.Add("RestoreLockedMode", "true");
```
Comment (dtivel) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: Remove.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs [hunk 2]
```diff
@@ -7405,5 +7405,153 @@ public async Task RestoreNetCore_IncompatiblePackageTypesFailRestore()
...
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    packageX);
+
+                // A -> B
+                projectA.Properties.Add("RestorePackagesWithLockFile", "true");
+                projectB.Properties.Add("PackageId", "CustomName");
+
+                //projectA.Properties.Add("RestoreLockedMode", "true");
```
Comment (dtivel) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: Remove.

**Review by dtivel — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs [hunk 3]
```diff
@@ -6124,23 +6124,23 @@ public async Task RestoreNetCore_InteropTypePackage()
         [Fact]
         public async Task RestoreNetCore_MultiTFM_ProjectToProject_PackagesLockFile()
         {
-            // Arrange
+            // Arrange	
             using (var pathContext = new SimpleTestPathContext())
             {
-                // Set up solution, project, and packages
+                // Set up solution, project, and packages
```
Comment (zivkan) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: trailing whitespace makes me sad 😢

**Review by zivkan — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs (hunk 3 above)
Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: I formatted the whole file and that's what the formatter gave me :)
I will go through it and remove any unnecessary changes.

Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: fixed.

**Pull request merged**